### ISSUE TITLE
Roslyn analyzers for bug-prone usage of DateTime/DateTimeOffset

### ISSUE THREAD

**Issue by timmydo** [dotnet/runtime#79105]
I recently ran into two separate bugs related to time handling in our code and thought roslyn analyzers might be able to assist.

The two instances were:

`DateTimeOffset.Parse("2020-01-02").ToString(...)`
and
`if (DateTimeOffset.TryParse(expiryString, out var expiry) && now < expiry) // expiry is a utc YYYY-MM-DD date`

in both of these usages, I should have used something like `DateTimeOffset.TryParse(expiryString, null, System.Globalization.DateTimeStyles.AssumeUniversal, out var expiry)`

I see this as similar to requiring explicit usage of culture objects in string comparisons. Is there an analyzer that could recommend the version of the function with the explicit default behavior request like we have for string comparison functions?

Semi-related #43956

**Comment by Clockwork-Muse** [comment]
> I see this as similar to requiring explicit usage of culture objects in string comparisons.

Yes, especially for one surprising reason - at least one culture uses a `YYYY-DD-MM` format.

> in both of these usages, I should have used something like 

You should be using 
```cs
DateTimeOffset.TryParse(expiryString, DateTimeFormatInfo.InvariantInfo, out var expiry)
```
... which is explicitly using a culture.

**Comment by timmydo** [comment]
Well, I'm glad my attempt at a fix is wrong if it makes the case for an analyzer stronger :)

**Comment by timmydo** [comment]
I don't think you should recommend invariant culture--you just need to recommend using an API where the culture is explicitly specified, similar to how string comparison API analyzers recommend you specify the culture explicitly.
